client/network/src/discovery.rs

```diff
@@ -514,7 +514,7 @@ impl NetworkBehaviour for DiscoveryBehaviour {
 			if !self.allow_private_ipv4 {
 				list_to_filter.retain(|addr| {
 					if let Some(Protocol::Ip4(addr)) = addr.iter().next() {
-						if addr.is_private() {
+						if !addr.is_global() {
```

Comment:
```suggestion
						if !IpNetwork::from(addr).is_global() {
```

So `is_global` is still unstable in the standard library, but we can use the `ip_network` library in order to have access to this method.

Reply:
It needs to be a match block:

```
match addr.iter().next() {
    Some(Protocol::Ip4(addr)) if !IpNetwork::from(addr).is_global() => return false,
    Some(Protocol::Ip6(addr)) if !IpNetwork::from(addr).is_global() => return false,
    _ => true
}
```